TFNetworkRecLayer.py

```diff
@@ -4670,10 +4670,13 @@ def _weights_time_axis_to_reduce(cls, weights, base):
     # SoftmaxOverSpatialLayer by default uses time_dim_axis.
     # We also should maybe check that this matches the base time dim axis.
     dyn_axes = weights.get_dynamic_axes()
+
     # for static time-dim
-    if weights.time_dim_axis not in dyn_axes:
+    if not dyn_axes or weights.time_dim_axis not in dyn_axes:
```

Comment:
This change (adding `not dyn_axes`) does not change anything. Why do you change it then? It's confusing now, because `not dyn_axes` is always False if the other condition is False, and if the other condition is True, it does not matter.